apps/rush-lib/src/logic/PublishUtilities.ts

```diff
@@ -72,17 +71,57 @@ export class PublishUtilities {
       }
     });
 
-    // For each requested package change, ensure downstream dependencies are also updated.
-    allChanges.packageChanges.forEach((change, packageName) => {
-      PublishUtilities._updateDownstreamDependencies(
-        change,
-        allChanges,
-        allPackages,
-        rushConfiguration,
-        prereleaseToken,
-        projectsToExclude
-      );
-    });
+    // keep resolving downstream dependency changes and version policy changes
+    // until no more changes are detected
+    let hasChanges: boolean;
+    do {
+      hasChanges = false;
+
+      // For each requested package change, ensure downstream dependencies are also updated.
+      allChanges.packageChanges.forEach((change, packageName) => {
+        hasChanges ||= PublishUtilities._updateDownstreamDependencies(
+          change,
+          allChanges,
+          allPackages,
+          rushConfiguration,
+          prereleaseToken,
+          projectsToExclude
+        );
```

Comment:
```suggestion
        const changeHasDownstreamChanges: boolean = PublishUtilities._updateDownstreamDependencies(
          change,
          allChanges,
          allPackages,
          rushConfiguration,
          prereleaseToken,
          projectsToExclude
        );
        hasChanges = hasChanges || changeHasDownstreamChanges;
```

```TypeScript
function y(): boolean {
  return true;
}
let x: boolean = false;
x ||= y();
```

compiles to 

```JavaScript
function y() {
    return true;
}
var x = false;
x || (x = y());
```

which isn't what you want here.